includes/wc-template-functions.php

```diff
@@ -335,27 +335,25 @@ function wc_body_class( $classes ) {
 
 	$classes[] = 'woocommerce-no-js';
 
-	add_action( 'wp_footer', 'wc_no_js' );
-
 	return array_unique( $classes );
 }
 
 /**
  * NO JS handling.
  *
+ * @param  array $classes Body Classes.
+ * @return array
+ *
  * @since 3.4.0
  */
-function wc_no_js() {
-	?>
-	<script type="text/javascript">
-		(function () {
-			var c = document.body.className;
-			c = c.replace(/woocommerce-no-js/, 'woocommerce-js');
-			document.body.className = c;
-		})()
-	</script>
-	<?php
+function wc_no_js( $classes ) {
```

Comment:
The parameter `$classes` needs to be optional. Otherwise, it can break things for plugins and/or themes that already call this function without any parameter.

Reply:
This function is called here only, not outside of plugin. Also this is filter for **body_class** which has one array parameter for classes.

Reply:
Well, I don't see anything that can stop plugin/theme devs to not use this function in their code. So, if they are free to do it, they might be doing it. No?

Reply:
Yes, they might use if they are free to do it.

Reply:
So if this PR gets merged and those people update WooCommerce, they will start getting a fatal error (something like `Too few arguments to function wc_no_js()`). AFAIK, these kinds of changes (the ones which can break a website) are only introduced in a major update.